### PR TITLE
Autosuggest artist and album

### PR DESCRIPTION
**Pull request by jvmsangkal**

### CODE REVIEW

File: app/scripts/components/staff/albums/modals/edit_album_modal.js [hunk 1]
```diff
@@ -3,12 +3,14 @@ var React = require('react'),
...
     RouteHandler = Router.RouteHandler,
 
+    AutosuggestText = require('components/common/ui/autosuggest_text'),
     InputText = require('components/common/ui/input_text'),
     SelectBox = require('components/common/ui/select_box'),
     UploadFile = require('components/common/ui/upload_file'),
     
     ArtistStore = require('stores/autocomplete/artist_store'),
     AlbumActions = require('actions/staff/album_actions'),
+    Autosuggest = require('react-autosuggest'),
```
Comment (sibayanjasper) [inline app/scripts/components/staff/albums/modals/edit_album_modal.js]: Remove this

File: app/scripts/components/staff/albums/modals/edit_album_modal.js [hunk 2]
```diff
@@ -42,6 +43,7 @@ var React = require('react'),
         render: function () {
             var image_btn_label = this.props.data.picture === null ? 'Add Image' : 'Change Image';
 
+
```
Comment (sibayanjasper) [inline app/scripts/components/staff/albums/modals/edit_album_modal.js]: Excess new line

File: app/scripts/components/music_provider/upload/bulk/music_info.js [hunk 3]
```diff
@@ -49,6 +49,7 @@ var React = require('react'),
             BulkUploadActions.update(this.state.active_track.track_id, data);
         },
         render: function () {
+
```
Comment (sibayanjasper) [inline app/scripts/components/music_provider/upload/bulk/music_info.js]: Excess new line

File: app/scripts/components/music_provider/tracks/modals/edit_track_modal.js [hunk 4]
```diff
@@ -7,6 +7,7 @@ var React = require('react'),
     ModalActions = require('actions/partials/modal_actions'),
     TrackActions = require('actions/music_provider/track_actions'),
     ArtistActions = require('actions/autocomplete/artist_actions'),
+    Autosuggest = require('react-autosuggest'),
```
Comment (sibayanjasper) [inline app/scripts/components/music_provider/tracks/modals/edit_track_modal.js]: Remove this.

File: app/scripts/components/staff/upload/bulk/autosuggest_album.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
+var React = require('react'),
```
Comment (sibayanjasper) [inline app/scripts/components/staff/upload/bulk/autosuggest_album.js]: Delete this file

File: app/scripts/components/staff/upload/bulk/autosuggest_artist.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
+var React = require('react'),
```
Comment (sibayanjasper) [inline app/scripts/components/staff/upload/bulk/autosuggest_artist.js]: Delete this file

File: app/scripts/components/staff/upload/bulk/music_info.js [hunk 7]
```diff
@@ -49,8 +49,6 @@ var React = require('react'),
             BulkUploadActions.update(this.state.active_track.track_id, data);
         },
         render: function() {
-            document.querySelector('input[id="' + this.state.active_track.status + '"]').checked = true;
```
Comment (sibayanjasper) [inline app/scripts/components/staff/upload/bulk/music_info.js]: Should this be deleted? or just deleted due to merge resolution?

Comment (jvmsangkal) [inline app/scripts/components/staff/upload/bulk/music_info.js]: baka sa merge. kailangan ata yan

Comment (sibayanjasper) [inline app/scripts/components/staff/upload/bulk/music_info.js]: @lyndeniece

File: app/scripts/components/common/ui/autosuggest_text.js [hunk 8]
```diff
@@ -1,51 +1,81 @@
 var React = require('react'),
+    _ = require('lodash'),
```
Comment (sibayanjasper) [inline app/scripts/components/common/ui/autosuggest_text.js]: unused

File: app/scripts/components/common/ui/autosuggest_text.js [hunk 9]
```diff
@@ -1,51 +1,81 @@
...
             var inputAttributes = {
-                    id: 'multiple-sections',
-                    placeholder: 'Search',
-                    name: this.props.name
-                };
+                    ref: 'input',
+                    id: this.props.id,
+                    value: '',
+                    placeholder: this.props.placeholder,
+                    required: this.props.required ? this.props.required : false
```
Comment (sibayanjasper) [inline app/scripts/components/common/ui/autosuggest_text.js]: @lyndeniece

File: app/scripts/components/common/ui/track_form.js [hunk 10]
```diff
@@ -15,17 +16,20 @@ var React = require('react'),
         cancelHandler: function () {
             ModalActions.dismiss();
         },
-        handleChange: function (id) {
-            ArtistActions.selectArtist(id);
-            this.refs.album.refs.select.clearCache();
-            this.refs.album.refs.select.clearValue();
+        handleChange: function (suggestion, event) {
```
Comment (sibayanjasper) [inline app/scripts/components/common/ui/track_form.js]: change name

File: app/scripts/components/music_provider/tracks/modals/edit_track_modal.js [hunk 11]
```diff
@@ -45,7 +43,6 @@ var React = require('react'),
                 });
         },
         componentDidMount: function () {
-            ArtistActions.selectArtist(this.refs.track_form.refs.artist.getValue() != undefined ? this.refs.track_form.refs.artist.getValue() : false);
             document.querySelector('input[id="' + this.props.data.status + '"]').checked = true;
```
Comment (sibayanjasper) [inline app/scripts/components/music_provider/tracks/modals/edit_track_modal.js]: move to track form

File: app/scripts/components/staff/tracks/modals/edit_track_modal.js [hunk 12]
```diff
@@ -45,10 +43,10 @@ var React = require('react'),
                 });
         },
         componentDidMount: function () {
-            ArtistActions.selectArtist(this.refs.track_form.refs.artist.getValue() != undefined ? this.refs.track_form.refs.artist.getValue() : false);
             document.querySelector('input[id="' + this.props.data.status + '"]').checked = true;
```
Comment (sibayanjasper) [inline app/scripts/components/staff/tracks/modals/edit_track_modal.js]: move to track form

File: app/scripts/components/staff/upload/bulk/music_info.js [hunk 13]
```diff
@@ -50,9 +48,9 @@ var React = require('react'),
         },
         render: function() {
             document.querySelector('input[id="' + this.state.active_track.status + '"]').checked = true;
```
Comment (sibayanjasper) [inline app/scripts/components/staff/upload/bulk/music_info.js]: move to track form

File: app/scripts/stores/autocomplete/artist_store.js [hunk 14]
```diff
@@ -13,10 +13,10 @@ var Reflux = require('reflux'),
             this.trigger();
         },
         get_completions: function (query, callback) {
-            music_api.artist.autocomplete(query, 10, function(err, result) {
+            music_api.artist.autocomplete(query, 5, function(err, result) {
                 if(err) {
```
Comment (sibayanjasper) [inline app/scripts/stores/autocomplete/artist_store.js]: not needed.
function should only have `callback(err, result.body.data)`

File: app/scripts/stores/autocomplete/artist_store.js [hunk 15]
```diff
@@ -24,10 +24,10 @@ var Reflux = require('reflux'),
             });
         },
         get_album_completions: function (query, callback) {
-            music_api.artist.autocomplete_album(artist_id, query, 10, function(err, result) {
+            music_api.artist.autocomplete_album(artist_id, query, 5, function(err, result) {
                 if(err) {
```
Comment (sibayanjasper) [inline app/scripts/stores/autocomplete/artist_store.js]: not needed.
function should only have `callback(err, result.body.data)`

**Pull request merged**